src/components/LineSeries.tsx

```diff
@@ -0,0 +1,61 @@
+import { extent } from 'd3-array';
+import { line } from 'd3-shape';
+import React, { CSSProperties, useEffect, useMemo, useState } from 'react';
+
+import { useDispatchContext, usePlotContext } from '../hooks';
+import type { LineSeriesProps, Series } from '../types';
+import { getNextId } from '../utils';
+
+interface LineSeriesRenderProps {
+  data: Series;
+  lineStyle?: CSSProperties;
+}
+
+export default function LineSeries(props: LineSeriesProps) {
+  const [id] = useState(`series-${getNextId()}`);
```

Comment:
```suggestion
  const [id] = useState(() => `series-${getNextId()}`);
```

To avoid generating new ids on every render